src/screens/TokenDetail.js

```diff
@@ -0,0 +1,207 @@
+   * It returns true if the transaction there is an input or output of the token
+   *
+   * @param {Object} tx Transaction data received in the websocket
+   *
+   * @return {boolean} True if should update the list, false otherwise
+   */
+  shouldUpdateList = (tx) => {
+    for (const input of tx.inputs) {
+      if (input.token === this.state.paramUID) {
+        return true;
+      }
+    }
+
+    for (const output of tx.outputs) {
+      if (output.token === this.state.paramUID) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /*
+   * Method called when updating the list with new data
+   * Call the API of token history
+   *
+   * @param {number} timestamp Timestamp reference of the pagination
+   * @param {string} hash Hash reference of the pagination
+   * @param {string} page Button clicked in the pagination ('previous' or 'next')
+   *
+   * @return {Promise} Promise to be resolved when new data arrives
+   */
+  updateListData = (timestamp, hash, page) => {
+    const promise = new Promise((resolve, reject) => {
+      hathorLib.walletApi.getTokenHistory(this.state.paramUID, TX_COUNT, hash, timestamp, page, (response) => {
+        resolve(response);
+      });
+    });
+    return promise;
+  }
+
+  render() {
+    if (this.state.errorMessage) {
+      return (
+        <div className="content-wrapper flex align-items-start">
+          <p className="text-danger">{this.state.errorMessage}</p>
+        </div>
+      )
+    }
+
+    if (!this.state.token) return null;
+
+    const configurationString = hathorLib.tokens.getConfigurationString(this.state.token.uid, this.state.token.name, this.state.token.symbol);
+
+    const getShortConfigurationString = () => {
+      const configArr = configurationString.split(':');
+      return `${configArr[0]}:${configArr[1]}...${configArr[3]}`;
+    }
+
+    const renderTokenInfo = () => {
+      return (
+        <div>
+          <p className="mt-3 mb-2"><strong>Total supply: </strong>{hathorLib.helpers.prettyValue(this.state.totalSupply)} {this.state.token.symbol}</p>
+          <p className="mt-2 mb-2"><strong>Can mint: </strong>{this.state.canMint ? <i className="fa fa-check ml-1" title="Can mint"></i> : <i className="fa fa-close ml-1" title="Can't mint"></i>}</p>
+          <p className="mt-2 mb-4"><strong>Can melt: </strong>{this.state.canMelt ? <i className="fa fa-check ml-1" title="Can melt"></i> : <i className="fa fa-close ml-1" title="Can't melt"></i>}</p>
+        </div>
+      );
+    }
+
+    return (
+      <div className="content-wrapper flex align-items-center">
+        <div className='d-flex flex-row align-items-start justify-content-between token-detail-top'>
+          <div className='d-flex flex-column justify-content-between mt-4'>
+            <div className='token-wrapper d-flex flex-row align-items-center mb-3'>
+              <p className='token-name mb-0'>
+                <strong>{this.state.token.name} ({this.state.token.symbol})</strong>
```

Comment:
I don't know where, but we should add a warning message that no one can trust the token without checking the token id. Also, the token id should be really visible.

I think we should also include a list of "conflicting" tokens to let users know about possible fraud.

Reply:
The alert message was added and we have talked about creating this conflicting tokens alert as an issue to be discussed in the future

Reply:
Good. Thanks!